Updated terminology

hannestschofenig opened the pull request:
Based on the discussions at IETF#103 (Bangkok) I have updated the terminology to align it with the terms used by the ITU-T, as described in this presentation https://datatracker.ietf.org/meeting/103/materials/slides-103-suit-liaison-statement-from-itu-t-sg17-00

hannestschofenig committed:
Updated terminology

Based on the discussions at IETF#103 (Bangkok) I have updated the terminology to align it with the terms used by the ITU-T, as described in this presentation https://datatracker.ietf.org/meeting/103/materials/slides-103-suit-liaison-statement-from-itu-t-sg17-00

henkbirkholz commented:
I really like this update to the draft!

A few comments (based on diff, pls excuse lack of context, if not included in diff):

"The firmware image can be confidentiality protected": Could this be a good place to differentiate confidentiality wrt data at rest (source) and data in motion (from source)?

"A single device may need to obtain more than one firmware image  and manifest to succesfully perform an update.": Could this be a good place to highlight that basically every device should be assumed to be a composite device (worst case a composite device with only one component). This generic assumption would prevent the assumption that pushing a single "blob" could be enough and encourage to take into account dependencies and the increasing complexity of even the smallest things?

"Status Tracker": Maybe the inclusion of term "campaign" beneficial to this section?

"Firmware Server: Entity [...]": I am not a big fan of the super generic term entity, but maybe that is not of importance here...

"Microcontroller": Maybe highlighting the characteristic of being "independent" and providing its own "computing context" is of relevance here?

"Client-initiated updates": Is the conceptual term of "polling" excluded here on purpose?

"LwM2M Firmware Update object": Is there a reason to get protocol specific in an architecture document at this point (there might be!).

Viele Grüße,

Henk

hannestschofenig committed:
Terminology updated based on comments from Henk

hannestschofenig commented:
I added new text and re-arranged things a bit. A few remarks:

"The firmware image can be confidentiality protected": Could this be a good place to differentiate confidentiality wrt data at rest (source) and data in motion (from source)?

[Hannes] I added a forward reference to the architecture section where e2e security is discussed. 

"A single device may need to obtain more than one firmware image and manifest to succesfully perform an update.": Could this be a good place to highlight that basically every device should be assumed to be a composite device (worst case a composite device with only one component). This generic assumption would prevent the assumption that pushing a single "blob" could be enough and encourage to take into account dependencies and the increasing complexity of even the smallest things?

[Hannes] I changed the text and included the definition of a Device where I point out that it has one or multiple MCUs. Let me know if the text is better now. 

"Status Tracker": Maybe the inclusion of term "campaign" beneficial to this section?

[Hannes] How do you think the term could be included here? 

"Firmware Server: Entity [...]": I am not a big fan of the super generic term entity, but maybe that is not of importance here...

[Hannes] OK. I changed the definition. 

"Microcontroller": Maybe highlighting the characteristic of being "independent" and providing its own "computing context" is of relevance here?

[Hannes] Can you make a suggestion? The current text is taken from Wikipedia and stripped down to the parts we care about. 

"Client-initiated updates": Is the conceptual term of "polling" excluded here on purpose?

[Hannes] I added the term. 

"LwM2M Firmware Update object": Is there a reason to get protocol specific in an architecture document at this point (there might be!).

[Hannes] I provided two examples and I believe I need to be more specific in the examples so they actually make sense.

d3zd3z commented on draft-ietf-suit-architecture.md:
```diff
@@ -90,20 +90,26 @@ When developing IoT devices, one of the most difficult problems
-functionality, or to re-configure the device.
+deployed in remote or inaccessible areas where manual 
+intervention is cost prohibitive or otherwise difficult. Updates 
+to the firmware of an IoT device is done to fix bugs in software, 
```
Updates...are done...

d3zd3z commented on draft-ietf-suit-architecture.md:
```diff
@@ -382,46 +385,43 @@ There are three broad classifications of update operating modes.
 The pre-authorisation step involves verifying whether the entity 
 signing the manifest is indeed authorized to perform an update. 
-The device must also determine whether it should fetching and 
+The firmware consumer must also determine whether it should fetching and 
```
...whether it should fetch and process ...

d3zd3z reviewed (changes requested):
Overall, good update, just a few grammar errors is all.

hannestschofenig committed:
Updates based on feedback from David Brown

hannestschofenig committed:
Submitted -04 draft version

hannestschofenig committed:
Submitted version -05 based on feedback from Jacob Beningo

Commit:
Architecture version -06

Proposed text based on discussions at the IETF meeting in Montreal. Added extra text about the status tracker, including a new figure describing an example use of the status tracker for firmware updates. Then, added text about the TEEP use case.
